### PR TITLE
- removed limit and skip from search params

### PR DESCRIPTION
**Pull request by bhartip2003**
- category added to search params
- api calling initiated using category from search params

### CODE REVIEW

File: src/pages/Product.jsx [hunk 1]
```diff
@@ -1,47 +1,59 @@
...
 
   useEffect(() => {
     dispatch(fetchCategory());
-    dispatch(fetchProducts({ limit: limit, skip: 0}));
+    dispatch(fetchProducts({ category: searchParamsCategory, limit: limit, skip: 0 }));
   }, []);
 
-  const handleButtonClick = (category) => {
-    setSelectedCategory(category);
+  const searchParamsCategory = searchParams.get("category");
```
[the remark this answers is not in the document]
Comment (bhartip2003) [reply src/pages/Product.jsx:29]: created searchParams file under constants folder and defined enum there.

File: src/components/product/InfiniteScroll.jsx [hunk 2]
```diff
@@ -21,14 +20,10 @@ const InfiniteScroll = ({ category }) => {
...
   const currPage = useSelector(currentPageSelector);
-  const [searchParams, setSearchParams] = useSearchParams();
   const productData = useSelector(productSelector);
   const loading = useSelector(loadingSelector);
 
-  useEffect(() => {
-    setSearchParams({ limit, skip });
-  }, [limit, skip]);
-
+  console.log("category in infinite scroll", category);
```
[the remark this answers is not in the document]
Comment (bhartip2003) [reply src/components/product/InfiniteScroll.jsx]: removed console

File: src/pages/Product.jsx [hunk 3]
```diff
@@ -53,7 +65,7 @@ const Product = () => {
       />
 
       {/* infinite scroll */}
-      <InfiniteScroll category={selectedCategory} />
+      <InfiniteScroll category={searchParamsCategory || null} />
```
[the remark this answers is not in the document]
Comment (bhartip2003) [reply src/pages/Product.jsx]: removed null as well

File: src/pages/Product.jsx [hunk 4]
```diff
@@ -1,47 +1,59 @@
...
+        category: searchParamsCategory,
         limit: limit,
         skip: 0,
       })
     );
+  }, [searchParamsCategory]);
+
+  const handleButtonClick = (category) => {
+    if (category) {
+      setSearchParams({ "category": category });
```
[the remark this answers is not in the document]
Comment (bhartip2003) [reply src/pages/Product.jsx]: used .set method to set category and .delete to remove category params

**Review by PoojaVishnoi18 — APPROVED** [review]

**Pull request merged**